lib/prompts/createProjectPrompt.ts

```diff
@@ -1,43 +1,63 @@
-// @ts-nocheck
-const fs = require('fs');
-const path = require('path');
-const {
+import fs from 'fs';
+import path from 'path';
+import {
   getCwd,
   sanitizeFileName,
   isValidPath,
   untildify,
-} = require('@hubspot/local-dev-lib/path');
-const { PROJECT_COMPONENT_TYPES } = require('../constants');
-const { promptUser } = require('./promptUtils');
-const { fetchFileFromRepository } = require('@hubspot/local-dev-lib/github');
-const { i18n } = require('../lang');
-const { logger } = require('@hubspot/local-dev-lib/logger');
-const { EXIT_CODES } = require('../enums/exitCodes');
-const {
+} from '@hubspot/local-dev-lib/path';
+import {
+  PROJECT_COMPONENT_TYPES,
   HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH,
   DEFAULT_PROJECT_TEMPLATE_BRANCH,
-} = require('../constants');
+} from '../constants';
+import { promptUser } from './promptUtils';
+import { fetchFileFromRepository } from '@hubspot/local-dev-lib/github';
+import { i18n } from '../lang';
+import { logger } from '@hubspot/local-dev-lib/logger';
+import { EXIT_CODES } from '../enums/exitCodes';
+import { RepoPath } from '@hubspot/local-dev-lib/types/Github';
 
 const i18nKey = 'lib.prompts.createProjectPrompt';
 
 const PROJECT_PROPERTIES = ['name', 'label', 'path', 'insertPath'];
 
-const hasAllProperties = projectList => {
+type ProjectsConfig = {
+  projects?: ProjectProperties[];
+};
+
+type ProjectProperties = {
+  name: string;
+  label: string;
+  path: string;
+  insertPath: string;
+};
+
+type CreateProjectPromptResponse = {
+  name: string;
+  dest: string;
+  template: ProjectProperties;
+};
+
+function hasAllProperties(projectList: ProjectProperties[]): boolean {
   return projectList.every(config =>
     PROJECT_PROPERTIES.every(p =>
       Object.prototype.hasOwnProperty.call(config, p)
     )
   );
-};
+}
 
-const createTemplateOptions = async (templateSource, githubRef) => {
+async function createTemplateOptions(
+  templateSource: string,
+  githubRef: string
+): Promise<ProjectProperties[]> {
   const hasCustomTemplateSource = Boolean(templateSource);
   const branch = hasCustomTemplateSource
     ? DEFAULT_PROJECT_TEMPLATE_BRANCH
     : githubRef;
 
-  const config = await fetchFileFromRepository(
-    templateSource || HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH,
+  const config: ProjectsConfig = await fetchFileFromRepository(
+    (templateSource as RepoPath) || HUBSPOT_PROJECT_COMPONENTS_GITHUB_PATH,
```

Comment:
You should be able to just change all of the types for `templateSource` to `RepoPath` and avoid casting here